src/coreclr/nativeaot/Runtime/amd64/GC.asm

```diff
@@ -19,3 +19,20 @@ LEAF_ENTRY xmmYmmStateSupport, _TEXT
 LEAF_END xmmYmmStateSupport, _TEXT
 
         end
+
+;; extern "C" DWORD __stdcall avx512StateSupport();
+LEAF_ENTRY avx512StateSupport, _TEXT
```

Comment:
This needs to be hooked up here: https://github.com/dotnet/runtime/blob/0a60d952587a405d00c58e327208ebc55ba9e7c0/src/coreclr/nativeaot/Runtime/startup.cpp#L166

Reply:
Looks like the AVX512 ISA checks need to be mirrored there as well.

Reply:
And the 'end' was in the wrong place :) . Have added the checks in startup.cpp